src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts

```diff
@@ -74,17 +74,14 @@ export class VehicleTechnicalRecordComponent implements OnInit, AfterViewInit {
 
     forms.forEach(form => DynamicFormService.updateValidity(form, errors));
 
-    errors.length ? 
-    this.errorService.setErrors(errors) 
-    : this.errorService.clearErrors();
+    errors.length ? this.errorService.setErrors(errors) : this.errorService.clearErrors();
 
     return forms.some(form => form.invalid);
   }
 
   handleFormState() {
-    const form = this.summary.sections
-    .map(section => section.form)
-    .concat(this.summary.dimensions.form, this.summary.weights.form);
+    const weightForm = this.summary.weightsTrlHgv?.form ? this.summary.weightsTrlHgv.form : this.summary.weights.form;
```

Comment:
Im thinking this check should probably be against the vehicle type on the evaluation, rather then checking if the form exists 😄

Reply:
I can do this, but the current tech record is an observable at this point, so I would have to subscribe every single time this method is called, which feels like adding massive overhead for very little gain. Unless you see a better way of doing it?